src/main/scala/rocket/Tile.scala

```diff
@@ -129,11 +131,14 @@ class RocketTile(tileId: Int)(implicit p: Parameters) extends LazyModule {
 
       ptwPorts ++= roccs.flatMap(_.io.ptw)
       uncachedArbPorts ++= roccs.flatMap(_.io.utl) // TODO no difference between io.autl and io.utl for now
-    }
 
-    val uncachedArb = Module(new ClientUncachedTileLinkIOArbiter(uncachedArbPorts.size)(icacheParams))
-    uncachedArb.io.in <> uncachedArbPorts
-    ucLegacy.module.io.legacy <> uncachedArb.io.out
+      val uncachedArb = Module(new ClientUncachedTileLinkIOArbiter(uncachedArbPorts.size)(icacheParams))
+      uncachedArb.io.in <> uncachedArbPorts
+      ucLegacy.module.io.legacy <> uncachedArb.io.out
+    } else {
+      ucLegacy.module.io.legacy.acquire.valid := Bool(false)
+      ucLegacy.module.io.legacy.grant.ready := Bool(false)
+    }
```

Comment:
Agreed, we should take the time to rethink RoCC entirely.  Probably need @yunsup @colinschmidt on board.